src/networking.c

```diff
@@ -2652,6 +2652,20 @@ void readQueryFromClient(connection *conn) {
         goto done;
     }
 
+    /* If during OOM and attempted to throttle clients traffic (in this state pending lazy free will
+     * soon release some memory), better pause early on writing big chunks to query-buffer, even
+     * before processing a complete command (otherwise it might be too late and too much payload copied
+     * already from the socket to query buffer). PROTO_BLOCKED_THRESHOLD is aimed to keep memory 
+     * consumption low of query_buf and on the other not too low - to reduce system calls and keep
+     * the pipeline busy with data to be processed once unblocked. */
+    if ((qblen > PROTO_BLOCKED_THRESHOLD) && (getPauseActionsPurposeEndTime(PAUSE_OOM_THROTTLE)))
+    {
+        if (!(c->flags & CLIENT_BLOCKED)) {
+            blockClient(c, BLOCKED_POSTPONE);
+        }
+        connSetReadHandler(c->conn,NULL);
```

Comment:
doesn't that limit all clients, not just ones that have been paused?
i.e. we don't throttle read commands, so we shouldn't block them or remove their read handler.
the only clients which we want to remove read handler for are ones that we decided not to process.

Reply:
you are right... i wanted to talk with you about this change. In case there is query_buf that grows 
behind some threshold I think it is safer to block it early on, even before processing a complete
command (read or write). The alternative of letting query_buf grow without any limit (during OOM) 
might be too risky. Note that before this PR, in case of FLUSHALL SYNC, during OOM or not, no
client could have write to the server.

Reply:
please review new commit. the explanation there might reflect my thought.

Reply:
reading your updated comment, still wonder.
so you say you remove the read handler for the `!BLOCKED`, because you wanna affect the ones that are gonna get blocked, earlier, before they manage to consume a big qbuf, but by that, you also affect reads, and PUBLISH, etc.

secondly IIRC, the reason why we wanted to remove the read handler was because we were afraid that in some cases the write command throttling can actually cause damage, e.g. a case where a pipeline of small SETs overwriting the same key again and again (not really consuming memory), will get blocked, and then the query buffer will grow (because of the combination of pipeline and our blocking).
i.e. we wanted that qbuf protection so that our throttling won't backfire on us.

lastly, IIRC you argued that in previously (before this PR), FLUSHALL would be completely blocking anyway, so this mechanism of throttling some reads, can't be considered a regression, i suppose that's right, but also it may not provide the full potential of this improvement.

Maybe we should focus on PUBLISH, since it's more likely to have a large query buffer than reads.
on the other hand, the real memory impact of PUBLISH, is actually not the query buffer, it's the output buffers it creates (which is amplified by the number of subscribers).